File: corpus/07_errors.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

func load(p string) (map[string]any, error) {
	data, err := os.ReadFile(p)
	if err != nil {
		return nil, err
	}
	var cfg map[string]any
	if err := json.Unmarshal(data, &cfg); err != nil {
		return nil, err
	}
	return cfg, nil
}

func main() {
	cfg, err := load("app.cfg")
	if err != nil {
		cfg = map[string]any{}
	}
	port, ok := cfg["port"]
	if !ok {
		port = 8080
	}
	fmt.Println(port)
}

File: corpus/07_errors.rs
use std::fs;

fn load(p: &str) -> Option<serde_json::Value> {
    let data = fs::read_to_string(p).ok()?;
    serde_json::from_str(&data).ok()
}

fn main() {
    let cfg = load("app.cfg").unwrap_or_else(|| serde_json::json!({}));
    let port = cfg.get("port").cloned().unwrap_or(serde_json::json!(8080));
    println!("{port}");
}
